src/components/ResourceList/components/FilterControl/components/DateSelector/tests/DateSelector.test.tsx

```diff
@@ -161,6 +172,48 @@ describe('<DateSelector />', () => {
     });
   });
 
+  describe('timezones adjustments', () => {
+    const date = '2019-01-01';
+
+    it('sets the correct selected date with timezone offset -540 on DatePicker and TextField', () => {
+      getTimezoneOffset.mockImplementation(() => -540);
+      const wrapper = mountWithAppProvider(
+        <DateSelector
+          {...mockDefaultProps}
+          filterValue={DateFilterOption.OnOrBefore}
+        />,
+      );
+
+      trigger(wrapper.find(TextField), 'onChange', date);
+      trigger(wrapper.find(TextField), 'onBlur');
+
+      const selectedDate = wrapper.find(DatePicker).prop('selected') as Date;
+      const selectedInputDate = wrapper.find(TextField).prop('value');
+
+      expect(selectedDate.toISOString()).toBe('2019-01-01T09:00:00.000Z');
```

Comment:
Does the timezone offset you mocked have an impact on `2019-01-01T09:00:00.000Z`? If yes, you should build this date string using the mocked offset.